guides/developer/crd.md

```diff
@@ -0,0 +1,211 @@
+    super(name);
+  }
+}
+```
+
+## Custom Spinnaker Naming Strategies
+
+Developers that want to use a custom naming strategy for ther CRDs should implement a `NamingStrategy`. For example,
+
+```
+@Component
+public class MyManifestNamer implements NamingStrategy<KubernetesManifest> {
+  @Override
+  public String getName() {
+    return "myManifestNamingStrategy";
+  }
+
+  @Override
+  public void applyMoniker(KubernetesManifest manifest, Moniker moniker) {
+    // Strategy for applying a Spinnaker `Moniker` to a Kubernetes
+    // manifest prior to deployment.
+  }
+
+  @Override
+  public Moniker deriveMoniker(KubernetesManifest manifest) {
+    // Strategy for deriving a Spinnaker `Moniker` from a Kubernetes
+    // manifest.
+  }
+}
+```
+
+This naming strategy can be referenced in a Kubernetes account config. For example:
+
+```
+kubernetes:
+  accounts:
+    - name: my-kubernetes-account
+      namingStrategy: myManifestNamingStrategy
+```
+
+Be careful - this naming strategy will be applied to all manifests manipulated by this account.
+
+## Custom Spinnaker UIs
+
+See the [reference repository](https://github.com/spinnaker/deck-customized) for developing Spinnaker UI customizations that
+should not be open sourced.
+
+Spinnaker's UI, Deck, has many points of integration to allow for per-cloud provider customization through the `CloudProviderRegistry`:
+
+```
+cloudProviderRegistryProvider.registerProvider('kubernetes', {
+  name: 'Kubernetes',
+  skin: 'v2',
+  ... // References to components and templates that are rendered per cloud provider throughout Deck.
+});
+```
+
+It's possible to override individual components and templates within the `CloudProviderRegistry` at build time
+using Deck's `OverrideRegistry`.
+However, these overrides are static and apply to every account using the UI implementation.
+
+The UIs for Spinnaker's Kubernetes V1 & V2 providers are implemented as `skin`s. A `skin` is a named UI implementation for a given provider, and
+is not strictly coupled to a backend implementation. By default, a Spinnaker Kubernetes account using the V1 provider implementation
+will use the V1 skin; an account using the V2 implementation will use the V2 skin.
+
+Developers can write custom skins for the Kubernetes provider by registering a new `skin` with the `CloudProviderRegistry`:
+
+```
+cloudProviderRegistryProvider.registerProvider('kubernetes', {
+  name: 'Kubernetes',
+  skin: 'myCustomizedUISkin',
+  ... // References to components and templates. These do not all need to be custom components -
+  ... // developers can mix in components from the V1 and V2 Kubernetes `skins`.
+});
+```
+
+This skin can be referenced in a Kubernetes account config. For example:
+
+```
+kubernetes:
+  accounts:
+    - name: my-kubernetes-account
+      skin: myCustomizedUISkin
+```
+
+A custom `skin` should be used in cases when some Kubernetes accounts should render the V1 or V2 `skin`, and others should use custom UI components.
```

Comment:
"Use a custom skin in cases where some Kubernetes account will render the V1 or V2...and other will use custom UI components." 
(avoid "should" if you can.)